kube/secret.go

```diff
@@ -50,9 +60,27 @@ func MakeSecrets(secrets model.CVMap, defaults map[string]string) ([]*apiv1.Secr
 	total := 0 // Accumulated size of the values stored in 'currentSecret'
 
 	for key, value := range secrets {
-		ok, strValue := configValue(value, defaults)
-		if !ok {
-			return nil, nil, fmt.Errorf("Secret '%s' has no value", key)
+		var strValue string
+		if createHelmChart {
+			switch {
+			case value.Generator == nil || value.Generator.Type != model.GeneratorTypePassword:
```

Comment:
The [switch statement] in go is actually more like an extended `if`/`else` chain:
> In an expression switch, the switch expression is evaluated and the case expressions, which need not be constants, are evaluated left-to-right and top-to-bottom; the first one that equals the switch expression triggers execution of the statements of the associated case; the other cases are skipped.

So as long as `value.Generator == nil` comes before `case value.Generator.Type` it's fine.  So you do end up with duplicate code, but the message can then be something like `Generator type %s is not supported for .Values.env.%s, required "%s"`.

[switch statement]: https://golang.org/ref/spec#Switch_statements